src/components/TaskGroup/TaskGroup.tsx

```diff
@@ -1,12 +1,70 @@
 import TaskItem from '../TaskItem/TaskItem';
 import { TaskItemGroupType } from '../../types/TaskItemGroupType';
-import styles from './TaskGroup.module.scss';
 import AddToDoSection from '../AddToDoSection/AddToDoSection';
+import styles from './TaskGroup.module.scss';
+import { TiDelete } from 'react-icons/ti';
+import { BiEdit } from'react-icons/bi';
+import IconButton from '../IconButton/IconButton';
+import clsx from 'clsx';
+import { useTaskStorage } from '../../hooks/useTaskStorage';
+import Title from '../Title/Title';
+import { useState, useEffect, useRef } from 'react';
+import useClickOutside from '../../hooks/useClickOutside';
 
 const TaskGroup = ({ taskList, groupTitle, groupId }: TaskItemGroupType) => {
+  const [isGroupEditable, setIsGroupEditbale] = useState(false);
+  const ref = useRef<HTMLInputElement>(null);
+  useClickOutside(
+    ref,
+    () => {
+      setIsGroupEditbale(false);
+    },
+    `ignoreClickOutside${groupId}`,
+  );
+  const { deleteTaskGroup, updateGroupTitle } = useTaskStorage();
+  const [title, setTitle] = useState(groupTitle);
+
+  useEffect(() => {
+    if (!isGroupEditable) {
+      updateGroupTitle(groupId, title);
+    }
+  }, [isGroupEditable]);
+
+  useEffect(() => {
+    if (ref?.current && isGroupEditable) {
+      ref.current.focus();
+    }
+  }, [isGroupEditable]);
+
   return (
     <div className={styles.TaskGroup}>
-      <h2>{groupTitle}</h2>
+      <div className={styles.taskGroupHeader}>
+        <Title
```

Comment:
Is this not editable on click on purpose ? (only editable on icon click)

Reply:
Yes thi is by purpose, I can change it if this will be more user freindly